### PR TITLE
See the bottom feature

### PR DESCRIPTION
**Pull request by rhit-daviscl**
Allows the User to see the bottom 3 cards

### CODE REVIEW

File: src/main/java/system/DrawDeck.java [hunk 1]
```diff
@@ -59,10 +59,36 @@ public List<Card> drawThreeCardsFromTop() {
...
+            cards.push(top.get(i));
+        }
 
         return top;
     }
 
+    public List<Card> drawThreeCardsFromBottom() {
+        if (cards.isEmpty()) {
+            throw new RuntimeException(I18n.getMessage("EmptyDrawDeckMessage"));
+        }
```
Comment (budak7273) [inline src/main/java/system/DrawDeck.java:72]: Is a test feasible for this case?

File: src/main/java/system/DrawDeck.java [hunk 2]
```diff
@@ -59,10 +59,36 @@ public List<Card> drawThreeCardsFromTop() {
         while (cards.size() > 0 && top.size() < maxTop) {
             top.add(cards.pop());
         }
+        for (int i = top.size() - 1; i >= 0; i--) {
+            cards.push(top.get(i));
+        }
```
Comment (budak7273) [inline src/main/java/system/DrawDeck.java:64]: I don't understand what this code is doing; it seems like it's putting back the cards that were taken?

**Review by budak7273 — CHANGES_REQUESTED** [review]
Looks good but I had 2 questions

File: src/main/java/system/DrawDeck.java (hunk 2 above)
Comment (rhit-daviscl) [reply src/main/java/system/DrawDeck.java:64]: We had the cards being added back to the top of the deck in a separate method chain that went on for a long time. While I was working on SeeTheEnd, I decided to make it so the DrawDeck is the thing that handles the adding on the top. So, the DrawDeck takes the cards from the top 3 cards from the top of the deck, then puts them right back on while keeping them saved in the list we return

File: src/main/java/system/DrawDeck.java (hunk 1 above)
Comment (rhit-daviscl) [reply src/main/java/system/DrawDeck.java:72]: This was a check that was also in draw3CardFromTop. So I just brought it over to this new method.

Comment (rhit-daviscl) [reply src/main/java/system/DrawDeck.java:72]: ya, one sec

Comment (rhit-daviscl) [reply src/main/java/system/DrawDeck.java:72]: It's pushed

File: src/main/java/system/DrawDeck.java (hunk 2 above)
Comment (rhit-daviscl) [reply src/main/java/system/DrawDeck.java:64]: image

File: src/main/java/system/DrawDeck.java [hunk 3]
```diff
@@ -59,10 +59,36 @@ public List<Card> drawThreeCardsFromTop() {
...
+        ArrayList<Card> bottom = new ArrayList<>();
+        Deque<Card> cardDeck = new LinkedList<>();
+        final int maxBottom = 3;
+
+        while (cards.size() - 2 >= maxBottom) {
+            cardDeck.add(cards.pop());
+        }
+        for (int i = 0; i <= cards.size(); i++) {
+            Card bottomCards = cards.pop();
+            System.out.println(bottomCards.getName() + " " + cards.size());
```
Comment (budak7273) [inline src/main/java/system/DrawDeck.java]:
```suggestion
```

**Review by budak7273 — CHANGES_REQUESTED** [review]

**Review by budak7273 — APPROVED** [review]

**Pull request merged**